terraform_token.go

```diff
@@ -101,46 +89,71 @@ func (b *tfBackend) terraformTokenRevoke(ctx context.Context, req *logical.Reque
 	if ok {
 		teamID, ok = teamIDRaw.(string)
 		if !ok {
-			return nil, fmt.Errorf("secret is missing team_id internal data")
+			return nil, fmt.Errorf("invalid value for team_id in secret internal data")
 		}
+
 	}
 
-	var data = map[string]interface{}{
-		"organization": organization,
-		"token_id":     tokenID,
-		"team_id":      teamID,
+	if isOrgToken(organization, teamID) {
+		// revoke org API token
+		if err := client.OrganizationTokens.Delete(ctx, organization); err != nil {
+			return nil, fmt.Errorf("error revoking organization token: %w", err)
+		}
+		return nil, nil
 	}
 
-	if err := b.walRollback(ctx, req, terraformTokenType, data); err != nil {
-		return nil, err
+	if isTeamToken(teamID) {
+		// revoke team API token
+		if err := client.TeamTokens.Delete(ctx, teamID); err != nil {
+			return nil, fmt.Errorf("error revoking team token: %w", err)
+		}
+		return nil, nil
+	}
+
+	// if we haven't returned yet, then the token is a user API token
+	tokenID := ""
+	tokenIDRaw, ok := req.Secret.InternalData["token_id"]
+	if ok {
+		tokenID, ok = tokenIDRaw.(string)
+		if !ok {
+			return nil, fmt.Errorf("secret is missing tokenID internal data")
+		}
+	}
+
+	if tokenID == "" {
+		return nil, fmt.Errorf("secret is missing tokenID internal data")
+	}
+
+	if err := client.UserTokens.Delete(ctx, tokenID); err != nil {
+		return nil, fmt.Errorf("error revoking user token: %w", err)
 	}
 	return nil, nil
 }
 
 func (b *tfBackend) terraformTokenRenew(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
 	roleRaw, ok := req.Secret.InternalData["role"]
 	if !ok {
-		return nil, fmt.Errorf("secret is missing token id internal data")
+		return nil, fmt.Errorf("secret is missing role internal data")
 	}
 
-	//get the credential entry
+	// get the role entry
 	role := roleRaw.(string)
-	cred, err := b.credentialRead(ctx, req.Storage, role)
+	roleEntry, err := b.getRole(ctx, req.Storage, role)
 	if err != nil {
-		return nil, errwrap.Wrapf("error retrieving role: {{err}}", err)
+		return nil, fmt.Errorf("error retrieving role: %w", err)
 	}
 
-	if cred == nil {
+	if roleEntry == nil {
 		return nil, errors.New("error retrieving role: role is nil")
 	}
 
 	resp := &logical.Response{Secret: req.Secret}
 
-	if cred.TTL > 0 {
-		resp.Secret.TTL = cred.MaxTTL
+	if roleEntry.TTL > 0 {
+		resp.Secret.TTL = roleEntry.MaxTTL
```

Comment:
Should this instead be `resp.Secret.TTL = roleEntry.TTL`?

Reply:
Indeed, thanks for the catch. Fixed in 6593fad